charts/datadog/templates/cluster-agent-advanced-confd-configmap.yaml

```diff
@@ -18,4 +18,41 @@ data:
     {{ $config | indent 4 | trim }}
 {{- end }}
 {{- end }}
+{{- if .Values.datadog.kubeStateMetricsCore.enabled }}
+  kubernetes_state_core.yaml.default: |-
+{{- if .Values.datadog.kubeStateMetricsCore.useClusterCheckRunners }}
+    cluster_check: true
+{{- end }}
+    init_config:
+    instances:
+      - collectors:
+{{- if .Values.datadog.kubeStateMetricsCore.collectSecretMetrics }}
+        - secrets
+{{- end }}
+        - nodes
+        - pods
+        - services
+        - resourcequotas
+        - replicationcontrollers
+        - limitranges
+        - persistentvolumeclaims
+        - persistentvolumes
+        - namespaces
+        - endpoints
+        - daemonsets
+        - deployments
+        - replicasets
+        - statefulsets
+        - cronjobs
+        - jobs
+        - horizontalpodautoscalers
+        - poddisruptionbudgets
+        - storageclasses
+        - volumeattachments
+{{- if .Values.datadog.kubeStateMetricsCore.useClusterCheckRunners }}
+        skip_leader_election: true
+{{- end }}
+        labels_as_tags:
+{{ .Values.datadog.kubeStateMetricsCore.labelsAsTags | toYaml | indent 10 }}
+{{- end }}
```

Comment:
Instead of duplicating the check config in `cluster-agent-advanced-confd-configmap.yaml` and `cluster-agent-confd-configmap.yaml`, we can define it once in a new file `templates/_default_kubernetes_state_core.yaml` and include it in both config maps. It would be easier to maintain.